custom_components/fraimic/library.py

```
@@ -207,6 +237,65 @@ async def async_update_image(
+            mimetypes.guess_type(filename)[0] or "application/octet-stream"
+        )
+
+        async def _rollback() -> bool:
+            rollback = await _async_await_cancellation_safe(
+                self.hass.async_add_executor_job(new_path.replace, old_path)
+            )
+            if isinstance(rollback.error, OSError):
+                rollback_err = rollback.error
+                raise HomeAssistantError(
+                    "Image rename failed and the original path could not be restored"
+                ) from rollback_err
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)